src/v/storage/log_reader.h

```diff
@@ -179,6 +179,8 @@ class log_reader final : public model::record_batch_reader::impl {
         _expected_next = _config.fill_gaps ? std::make_optional<model::offset>(
                                                _config.start_offset)
                                            : std::nullopt;
+        _last_base = {};
```

Comment:
Good point. I'll put something up.

In fact, looking at this it seems that the `if (config.abort_source) {` part in the constructor actually also needs to be part of reset: currently aborts will be ignored on cached readers (or possibly even something worse if for some reason the abort is triggered after the reader is returned to the cache).

Reply:
https://github.com/redpanda-data/redpanda/commit/354b484f13ed9f5c734fa2ae21c272f7c92b69e7 shares code between the ctor and reset_config with the private `reset()` method, though I don't think it would have caught this specific case because `_last_base` isn't explicitly initialized in the constructor, rather it is just default constructed.

It also moves the abort source handling into this shared code, fixing the issue described above.